### PR TITLE
[fleet] use custom tracer

### PR DESCRIPTION
**Pull request by raphaelgavache**
- Remove the dependency on go-tracer, immediately reducing downloader and installer by 12MB
- Removed reference of the trace-agent in the telemetry. The reference `Endpoint` result in importing many trace-agent dependencies

Size change:

Before
```
❯ ls -lh ./bin/installer
total 144368
-rwxr-xr-x  1 raphael.gavache  staff    19M 17 déc 22:42 downloader
-rw-r--r--  1 raphael.gavache  staff    52M 17 déc 22:42 install-databricks.sh
```

After
```
ls -lh ./bin/installer
total 55736
-rwxr-xr-x  1 raphael.gavache  staff   7,3M 17 déc 22:43 downloader
-rw-r--r--  1 raphael.gavache  staff    20M 17 déc 22:43 install-databricks.sh
```

### CODE REVIEW

File: pkg/fleet/telemetry/http_wrapper.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	span.SetTag("http.status_code", strconv.Itoa(res.StatusCode))
+	if res.StatusCode >= 400 {
+		span.SetTag("http.errors", res.Status)
+	}
+	return res, err
+}
+
+// urlFromRequest returns the URL from the HTTP request. The URL query string is included in the return object iff queryString is true
+// See https://docs.datadoghq.com/tracing/configure_data_security#redacting-the-query-in-the-url for more information.
+func urlFromRequest(r *http.Request, queryString bool) string {
```
Comment (BaptisteFoy) [inline pkg/fleet/telemetry/http_wrapper.go:59]: nit: maybe we should get rid of `queryString` given that it's always `true`?

File: pkg/fleet/telemetry/span.go [hunk 2]
```diff
@@ -7,20 +7,146 @@
...
+}
+
+func newSpan(name string, parentID, traceID uint64) *Span {
+	var noParent bool
+	if traceID == 0 {
+		traceID = rand.Uint64()
+		if !headSamplingKeep(name, traceID) {
+			traceID = dropTraceID
+		}
+		noParent = true
```
Comment (BaptisteFoy) [inline pkg/fleet/telemetry/span.go:40]: Shouldn't this be set if `parentID == 0` instead?

**Review by BaptisteFoy — APPROVED** [review]

File: pkg/internaltelemetry/traces.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	// Resource is the name of the resource this span represents.
+	Resource string `json:"resource"`
+	// TraceID is the ID of the trace to which this span belongs.
+	TraceID uint64 `json:"trace_id"`
+	// SpanID is the ID of this span.
+	SpanID uint64 `json:"span_id"`
+	// ParentID is the ID of the parent span.
+	ParentID uint64 `json:"parent_id"`
+	// Start is the start time of this span in nanoseconds since the Unix epoch.
+	Start int64 `json:"start""`
```
Comment (BaptisteFoy) [inline pkg/internaltelemetry/traces.go:30]:
```suggestion
	Start int64 `json:"start"`
```

File: pkg/fleet/telemetry/span.go (hunk 2 above)
Comment (raphaelgavache) [reply pkg/fleet/telemetry/span.go:40]: both work, it is nicer with parentID

**Review by BaptisteFoy — APPROVED** [review]

**Review by clarkb7 — APPROVED** [review]

**Pull request merged**